game/ui/general/savemenu.h

```diff
@@ -36,11 +36,12 @@ class SaveMenu : public Stage
 	void loadWithWarning(sp<Control> parent);
 	// these functions will display prompt to execute action
 	void tryToLoadGame(sp<Control> slotControl);
-	void tryToSaveGame(const UString &textEdit, sp<Control> parent);
+	void tryToSaveGame(const UString &textEdit, const sp<Control> parent);
 	void tryToDeleteSavedGame(sp<Control> &control);
 
 	// Opens pop-up asking the user if wants to override existing saved game
-	void askUserIfWantToOverrideSavedGame(sp<SaveMetadata> saveMetadata);
```

Comment:
I'm not sure if SaveMetadata needs to be wrapped in an sp<> - it's just a stack of strings, copying it around would be easier and simpler?

Reply:
Making getSaveGameIfExists() et al return an optional<SaveMetadata> might be more obvious in it's intent?

Reply:
Could you please show a suggestion regarding how to return this as an optional?

Reply:
Again it's probably not an issue for this PR, as it's already using an sp<>, but it doesn't seem like it's necessary.

For things that *may* return a SaveMetadata, instead of using an sp<SaveMetadata> and comparing it against nullptr, a std::optional<SaveMetadata> is probably a more fitting abstraction to how it's actually used.

e.g. you can do something like:
```
std::optional<SaveMetadata> get_save()
{
  if (save_exists)
    return SaveMetadata{save, metadata,. arguments};
  else
   return {}; // Returns an "empty" optional
}

void do_other_things_with_save(const SaveMetadata &metadata) // You *must* have a valid SaveMetadata already, so no need for this to be std::optional<SaveMetadata>.
{
....whatever
}

void doThingWithSave()
{
  auto optional_metadata= getSave();
  if (optional_metadata)
  {
     // get_save() returned an optional<> containing a SaveMetadata
     // operator* returns the underlying object of an optional<>
     const auto &metadata = *metadata;
     do_other_things_with_save(metadata);
  }
  else
  {
    // get_save() returned  an "empty" optional
  }
}
```

an sp<> is a reference counted pointer to a heap allocation that then contains the SaveMetadata (or a nullptr) - but that seems unnecessary if the SaveMetadata doesn't actually *need* shared ownership reference counting....

Reply:
Updated `SaveManager::getSaveGameIfExists` return type to `std::optional<SaveMetadata>`:

```c++
std::optional<SaveMetadata> SaveManager::getSaveGameIfExists(const UString &name) const
{
	const auto saveList = getSaveList();
	const auto it =
	    std::find_if(saveList.begin(), saveList.end(),
	                 [&name](const SaveMetadata &obj) { return obj.getName() == name; });

	if (it != saveList.end())
	{
		return *it;
	}

	return {};
}
```

Also updated `SaveMenu::askUserIfWantToOverrideSavedGame` parameters to use `const SaveMetadata &saveMetadata` instead of shared pointer.

Thanks for the help, really appreciate it!